tests/components/modbus/test_init.py

```diff
@@ -601,6 +606,79 @@ async def test_duplicate_entity_validator_with_climate(do_config) -> None:
     assert len(do_config[0][CONF_CLIMATES]) == 1
 
 
+@pytest.mark.parametrize(
+    "do_config",
+    [
+        [
+            {
+                CONF_NAME: TEST_ENTITY_NAME,
+                CONF_ADDRESS: 1,
+                CONF_HVAC_MODE_REGISTER: {
+                    CONF_ADDRESS: 119,
+                    CONF_HVAC_MODE_VALUES: {
+                        CONF_HVAC_MODE_COOL: 0,
+                        CONF_HVAC_MODE_HEAT: 1,
+                    },
+                    CONF_HVAC_MODE_TT_REG_BY_VALUES: {
+                        CONF_HVAC_MODE_COOL_TT_REG: 119,
+                        CONF_HVAC_MODE_HEAT_TT_REG: 100,
+                    },
+                },
+                CONF_FAN_MODE_REGISTER: {
+                    CONF_ADDRESS: 121,
+                },
+            },
+        ],
+        [
+            {
+                CONF_NAME: TEST_ENTITY_NAME + " 2",
+                CONF_ADDRESS: 2,
+                CONF_HVAC_MODE_REGISTER: {
+                    CONF_ADDRESS: 118,
+                    CONF_HVAC_MODE_VALUES: {
+                        CONF_HVAC_MODE_COOL: 0,
+                        CONF_HVAC_MODE_HEAT: 1,
+                    },
+                    CONF_HVAC_MODE_TT_REG_BY_VALUES: {
+                        CONF_HVAC_MODE_COOL_TT_REG: 119,
+                        CONF_HVAC_MODE_HEAT_TT_REG: 100,
+                    },
+                },
+                CONF_FAN_MODE_REGISTER: {
+                    CONF_ADDRESS: 100,
+                },
+            },
+        ],
+        [
+            {
+                CONF_NAME: TEST_ENTITY_NAME + " 3",
+                CONF_ADDRESS: 3,
+                CONF_HVAC_MODE_REGISTER: {
+                    CONF_ADDRESS: 118,
+                    CONF_HVAC_MODE_VALUES: {
+                        CONF_HVAC_MODE_COOL: 0,
+                        CONF_HVAC_MODE_HEAT: 1,
+                    },
+                    CONF_HVAC_MODE_TT_REG_BY_VALUES: {
+                        CONF_HVAC_MODE_COOL_TT_REG: 119,
+                        CONF_HVAC_MODE_HEAT_TT_REG: 117,
+                    },
+                },
+                CONF_HVAC_ONOFF_REGISTER: 117,
+                CONF_FAN_MODE_REGISTER: {
+                    CONF_ADDRESS: 100,
+                },
+            },
+        ],
+    ],
+)
+async def test_climate_conflict_addresses(do_config) -> None:
+    """Test conflicts among the addresses of target temp."""
+    check_hvac_target_temp_registers(do_config[0])
+    regs = do_config[0].get(CONF_HVAC_MODE_REGISTER)
+    assert len(regs[CONF_HVAC_MODE_TT_REG_BY_VALUES]) == 1
+
```

Comment:
All options must be tested, also a test with duplicate options, and tests with duplicate addresses.

This integration have 100% test coverage, so you also need to secure that all lines are tested.

Reply:
Thanks @janiversen for the review. 
The code has the same coverage of the whole previous code. (100%)
Duplicated registers into the map cannot exist because I used a set.  I try to answer to your requests point by point. 
Thanks again.